gtsam/discrete/DecisionTreeFactor.cpp

```diff
@@ -134,17 +134,33 @@ namespace gtsam {
     return boost::make_shared<DecisionTreeFactor>(dkeys, result);
   }
 
+  /* ************************************************************************* */
+  std::vector<std::pair<DiscreteValues, double>> DecisionTreeFactor::enumerate() const {
+    // Get all possible assignments
+    std::vector<std::pair<Key, size_t>> pairs;
+    for (auto& key : keys()) {
+      pairs.emplace_back(key, cardinalities_.at(key));
+    }
+    std::vector<std::pair<Key, size_t>> rpairs(pairs.rbegin(), pairs.rend());
```

Comment:
Could you explain why it has to be reversed?

Reply:
Added comment `// Reverse to make cartesianProduct output a more natural ordering.`